File: app/src/main/java/com/foxek/simpletimer/data/database/IntervalDAO.java
package com.foxek.simpletimer.data.database;

import com.foxek.simpletimer.data.model.Interval;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import io.reactivex.Flowable;
import io.reactivex.Single;

@Dao
public interface IntervalDAO {

    @Insert
    void add(Interval interval);

    @Query("DELETE FROM Interval WHERE id IS :id AND trainingID IS :workoutId")
    void delete(int id, int workoutId);

    @Update
    void update(Interval interval);

    @Query("SELECT * FROM Interval WHERE id IS :id AND trainingID IS :workoutId")
    Single<Interval> getById(int id, int workoutId);

    @Query("SELECT COUNT(*) FROM Interval WHERE trainingID IS :workoutId")
    Single<Integer> size(int workoutId);

    @Query("SELECT * FROM Interval WHERE trainingID IS :workoutId")
    Flowable<List<Interval>> getAll(int workoutId);

    @Query("SELECT MAX(id)  FROM Interval")
    Single<Integer> getLastId();
}

File: app/src/main/java/com/foxek/simpletimer/data/database/IntervalDAO.kt
package com.foxek.simpletimer.data.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.foxek.simpletimer.data.model.Interval

import io.reactivex.Flowable
import io.reactivex.Single

@Dao
interface IntervalDAO {

    @Insert
    fun add(interval: Interval)

    @Query("DELETE FROM Interval WHERE id IS :id AND trainingID IS :workoutId")
    fun delete(id: Int, workoutId: Int)

    @Update
    fun update(interval: Interval)

    @Query("SELECT * FROM Interval WHERE id IS :id AND trainingID IS :workoutId")
    fun getById(id: Int, workoutId: Int): Single<Interval>

    @Query("SELECT COUNT(*) FROM Interval WHERE trainingID IS :workoutId")
    fun size(workoutId: Int): Single<Int>

    @Query("SELECT * FROM Interval WHERE trainingID IS :workoutId")
    fun observeAll(workoutId: Int): Flowable<List<Interval>>

    @Query("SELECT * FROM Interval WHERE trainingID IS :workoutId")
    fun getAll(workoutId: Int): Single<List<Interval>>

    @Query("SELECT MAX(id)  FROM Interval")
    fun getLastId(): Single<Int>
}
